raiden-network/light-client
# Improve first and non-first initial sync times on testnet and mainnet, reduce number of ETH RPC requests

Fixes #2566
Fixes #2569 

**Short description**
This PR optimizes some filters for first, initial and on-block syncs/events monitoring.
- `getLogsByChunk$` now can fetch events from arrays of `address`es, allowing a single request to be used to monitor multiple contracts at the same time
- The above is used in order to remove a long standing workaround in code: now, all TokenNetworks are scanned in search of channels with us, to decide on TNs "of interest" to be monitored
  - Besides this, we also always monitors the first 2 TNs found, in order to avoid this re-scan in cases no channels are found, e.g. on mainnet, which did slow down the non-first initial syncs in case no TN got monitored for that client.
- `fromEthersEvent` can now properly report when its first/past event scanning phase is completed, so we can better report it at `synced` promise and `deps.init$` subject
  - Also, it can take a `blockNumber$` observable to use, instead of listening `provider.on('block')`, which does start the block polling. This was done in order to delay this polling until `synced` on `initNewBlockEpic` and passing the `block/new` action-based observable whenever this function is used.
- On TokenNetwork scanning, we now rely on `fromEthersEvent` working properly, and resume `fromBlock` since latest seen block, which reduces by a lot the range of blocks on non-first initial syncs.
- `pfsList` is replaced by `servicesValid`, which is persisted on `state.services` and resumed properly on next initial sync, instead of scanning the whole deployment range again, which was one of the other bottlenecks on initial sync times.
  - a db migration is included to include this property on already initialized states
- waiting for `messageGlobalSend` responses on `init$`/`synced`, waits for init-time `PFSFeeUpdate` global messages to be sent before considering node `synced`
- **TODO on a follow up PR**: monitors all TNs by default using a single filter with all TN's addresses, possibly also using this same filter/request (once per block) to perform all events fetching on the LC, not only TN's. This will reduce as much as possible the number of requests per time-unit on the LC, which will optimize on Infura usecases. Possibly, we can do with a single getLogs request per block!

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Initialize a CLI & dApp client on a clean state, on testnet and mainnet, and notice the `raiden/synced`'s `payload.tookMs` value;
2. Restart it, and see this time drop to < 5s if there're channels; <1s if there's no channel (due to the global messages)
